ibazel/command/command.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package command
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"syscall"
+
+	"github.com/bazelbuild/bazel-watcher/bazel"
+)
+
+var execCommand = exec.Command
+
+type Command interface {
+	Start()
+	Terminate()
+	NotifyOfChanges()
+	IsSubprocessRunning() bool
+}
+
+func run(b bazel.Bazel, target string, args []string) *exec.Cmd {
+	tmpfile, err := ioutil.TempFile("", "bazel_script_path")
+	if err != nil {
+		fmt.Print(err)
+	}
+	// Close the file so bazel can write over it
+	if err := tmpfile.Close(); err != nil {
+		fmt.Print(err)
+	}
+
+	// Start by building the binary
+	b.Run("--script_path="+tmpfile.Name(), target)
+
+	targetPath := tmpfile.Name()
+
+	// Now that we have built the target, construct a executable form of it for
+	// execution in a go routine.
+	cmd := execCommand(targetPath, args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	// Set a process group id (PGID) on the subprocess. This is
+	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+
+	// Start run in a goroutine so that it doesn't block watching for files that
+	// have changed.
+	if err := cmd.Start(); err != nil {
+		fmt.Printf("Error starting process: %v\n", err)
+	}
+
+	return cmd
+}
+
+func subprocessRunning(cmd *exec.Cmd) bool {
```

Comment:
This operates on an exec.Cmd instead of a Command and things that implement Command (currently only in this file) are the only things in the repo that interact with exec.Cmd. If it is necessary elsewhere we can extract it